src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,135 @@
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+
+import duke.DukeException;
+
+
+/**
+ * Commandline GUI class to handle all the printing in the application.
+ */
+
+public class Ui {
+    public static final String INDENT = "   ";
+    public static final String BAR = "____________________________________________________________";
+    public static final String MESSAGE_BYE = " Bye. Hope to see you again soon!";
+    private static final String FIRST = " Hello! I'm Pluiexo";
+    private static final String SECOND = " What can I do for you?";
+    private static final String[] GREET = new String[]{BAR, FIRST, SECOND, BAR};
+    private static final String[] BYE = new String[]{BAR, MESSAGE_BYE, BAR};
+
+    private final BufferedReader READER = new BufferedReader(new InputStreamReader(System.in));
+
+    /**
+     * Creates the GUI handler.
+     */
+    public Ui() {
+
+    }
+
+    /**
+     * Displays a break line bar of dashes with indentation.
+     */
+    public void showLine() {
+        System.out.println(INDENT + BAR);
+    }
+
+    /**
+     * Draws an empty line in output.
+     */
+    public void spacer() {
+        System.out.println();
+    }
+
+    /**
+     * Displays the welcome greeting of the chatbot.
+     */
+    public void showWelcome() {
+        for (String l : GREET) {
+            System.out.println(INDENT + l);
+        }
+    }
+
+    /**
+     * Scans the next line for instructions to the chatbot.
+     *
+     * @return String value of the user input.
+     * @throws DukeException
+     */
+    public String readInstructions() throws DukeException {
+        String test = "";
+        try {
+            test = READER.readLine();
+        } catch (IOException e) {
+            throw new DukeException("Gigaboom");
+        }
+        return test;
+    }
+
+    /**
+     * Prints to the output console of all the items in the list object.
+     *
+     * @param data An ArrayList of string to print everything to the output with indentation.
+     */
+    public void printOutput(ArrayList<String> data) {
+        for (String item : data) {
+            System.out.println(INDENT + item);
+        }
+    }
+
+    /**
+     * Prints to the output console of all the items in the array object.
+     *
+     * @param data A String array to print everything to the output with indentation.
+     */
+    public void printOutput(String[] data) {
+        for (String item : data) {
+            System.out.println(item);
+        }
+    }
+
+    /**
+     * Decides what error message to show based on the DukeException error message
+     *
+     * @param error DukeException error message.
+     */
+    public void showError(String error) {
+        //Used when your user does not know how to use the application
+        String errorMsg;
+        switch (error) {
+        case "description":
+            errorMsg = " OOPS!!! The description of this command cannot be empty.";
```

Comment:
in my personal opinion, maybe can change to errorMessage instead of errorMsg to make it even clearer but i am pretty sure errorMsg works too